source/common/network/connection_impl.cc

```diff
@@ -205,6 +212,10 @@ void ConnectionImpl::closeSocket(ConnectionEvent close_type) {
   connection_stats_.reset();
 
   file_event_.reset();
+
+  // Shutdown the socket so the other side can read any data in flight.
+  Api::OsSysCallsSingleton::get().shutdown(ioHandle().fd(), ENVOY_SHUT_WR);
```

Comment:
This is a functional change on Linux, right? Can you revert this and we can do an independent PR on this and discuss?

Reply:
Agreed, thanks @mattklein123 , addressed and then remerging master to possibly pass coverage.